scripts/device-common-install/libexec/device_hw_validation.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+is_expected_hdd() {
+  # $1 = Expected value, $2 = Tolerance
+  if [ $# -lt 2 ]; then
+    return 1
+  fi
+  hdd_lower_bound=$(expr "$1" - "$2")
+  hdd_upper_bound=$(expr "$1" + "$2")
+  hdd_size="$(lsblk /dev/sda | head -n 2 | tail -n 1 | awk '{print $4}' | sed 's/[A-Z]//g')"
+
+  test "$hdd_size" -ge "$hdd_lower_bound" && test "$hdd_size" -le "$hdd_upper_bound"
+}
+
+is_expected_ram() {
+  # $1 = Expected value, $2 = Tolerance
+  if [ $# -lt 2 ]; then
+    return 1
```

Comment:
Error message to stderr required here. (e.g.: 1>&2 echo "ERROR: My error!"). Why not instead take the arguments the following way instead:

```bash
local expected_value
expected_value="${1?ERROR: My error message}"
```